src/AmalgamatorSession.ts

```diff
@@ -488,6 +498,56 @@ export class AmalgamatorSession extends LoggingDebugSession {
         }
     }
 
+    protected async disassembleRequest(
+        response: DebugProtocol.DisassembleResponse,
+        args: DebugProtocol.DisassembleArguments
+    ): Promise<void> {
+        if (args.memoryReference) {
+            response.body = {
+                instructions: [],
+            };
+            try {
+                this.addressMap.forEach(
+                    (
+                        [instructionPointerReference, startAdress, endAddress],
+                        childDapID
+                    ) => {
+                        if (
+                            args.memoryReference ===
+                                instructionPointerReference ||
+                            args.memoryReference === startAdress ||
+                            args.memoryReference === endAddress
+                        ) {
+                            this.childDapIndex = childDapID;
+                        }
+                    }
+                );
+                const disassemble = await this.childDaps[
+                    this.childDapIndex as number
```

Comment:
This type casting looks wrong to me. I assume it is here because childDapIndex is optional. If childDapIndex is not set, that should simply be an error.

Reply:
I have updated it.